src/WebJobs.Script/Description/Node/NodeFunctionInvoker.cs

```diff
@@ -298,13 +296,25 @@ protected override void OnScriptFileChanged(object sender, FileSystemEventArgs e
                 { "bind", bind }
             };
 
+            // This is the input value that we will use to extract binding data.
+            // Since binding data extraction is based on JSON parsing, in the
+            // various conversions below, we set this to the appropriate JSON
+            // string when possible.
+            object bindDataInput = input;
+
             if (input is HttpRequestMessage)
             {
                 // convert the request to a json object
                 HttpRequestMessage request = (HttpRequestMessage)input;
-                var requestObject = CreateRequestObject(request);
+                string rawBody = null;
+                var requestObject = CreateRequestObject(request, out rawBody);
                 input = requestObject;
 
+                if (rawBody != null)
+                {
+                    bindDataInput = rawBody;
```

Comment:
This is the actual fix. Previously the binding data code we already have in place wasn't extracting the data because input was of type HttpRequestMessage in these cases.